onewire_driver.cpp

```diff
@@ -0,0 +1,266 @@
+     *  @param  fn:     function pointer to execute
+     *  @param  place:  additinal text to warning message
+     */
+
+    template <typename F>
+    inline void SuppressExceptions(F && fn, const char * place)
+    {
+        try {
+            fn();
+        } catch (const exception & e) {
+            LOG(Warn) << "Exception at " << place << ": " << e.what();
+        } catch (...) {
+            LOG(Warn) << "Unknown exception in " << place;
+        }
+    }
+
+    /*  @brief  Comparing two unordered vector and returns their difference
+     *  @param  first:  base vector reference
+     *  @param  second: secondary vector reference
+     *  @param  result: result vector reference
+     *  @note   first - second = result
+     */
+
+    template<typename T>
+    void UnorderedVectorDifference(const vector<T> &first, const vector<T>& second, vector<T> & result)
+    {
+        for (auto & el_first: first) {
+            bool found = false;
+            for (auto & el_second: second) {
+                if (el_first == el_second) {
+                    found = true;
+                    break;
+                }
+            }
+
+            if (!found) {
+                result.push_back(el_first);
+            }
+        }
+    }
+    
+}
+
+/*  @brief  class constructor, uses delegated constructor
+ *  @param  mqttDriver: mqtt handler object, required for handling MQTT   
+ *  @note   If functions is called, default polling interval is used
+ */
+
+TOneWireDriver::TOneWireDriver (const WBMQTT::PDeviceDriver & mqttDriver) 
+{
+    TOneWireDriver (mqttDriver, DEFAULT_POLL_INTERVALL_MS);
+}
+
+/*  @brief  class constructor, it should be called at each object creation (other constructors should call it too)
+ *  @param  mqttDriver:     mqtt handler object, required for handling MQTT   
+ *  @param  p_intvall_ms:   polling intervall in microseconds
+ *  @note   mqtt device is creater here however control channels are not
+ */
+
+
+TOneWireDriver::TOneWireDriver (const WBMQTT::PDeviceDriver & mqttDriver, int p_intvall_ms) : MqttDriver(mqttDriver), poll_intervall_ms(p_intvall_ms), Active(false)
+{
+
+    if (p_intvall_ms <= 0) {
+        throw invalid_argument("polling intervall must be greater than zero");
+    }
+
+    try {
+        auto tx = MqttDriver->BeginTx();
+
+        DeviceP = tx->CreateDevice(TLocalDeviceArgs{}
+            .SetId(Name)
+            .SetTitle("1-wire Thermometers")
+            .SetIsVirtual(true)
+            .SetDoLoadPrevious(false)
+        ).GetValue();
+
+    } catch (const exception & e) {
+        LOG(Error) << "Unable to create W1 driver: " << e.what();
+        throw;
+    }
+
+    EventHandlerHandle = mqttDriver->On<TSyncEvent>([](const TSyncEvent & event){
+        
+    });
+}
+
+/*  @brief class destructor function, clears all the MQTT devices and channels
+ */
+
+TOneWireDriver::~TOneWireDriver()
+{
+    if (EventHandlerHandle) {
```

Comment:
You won't have EventHandlerHandle, so you will need to find another way to know if Clear was already called or not.